nix-tools-default.nix

```diff
@@ -0,0 +1,37 @@
+commonLib: # the iohk-nix commonLib, that provides access to the pinned packages
+pkgs-path: # the path to th local pkgs.nix file for nix-tools that imports all
+           # haskell specific data.
+
+{ system ? builtins.currentSystem
+, crossSystem ? null
+, config ? {}
+, pkgs ? commonLib.getPkgs { inherit system crossSystem config; }
+}:
+with builtins; with pkgs.lib;
+let  nix-tools = import pkgs-path { nixpkgs = _: pkgs; };
+in {
+    _lib = commonLib;
+
+    # This will allow us to build
+    # nix-tools.libs.cardano-chain to obtain all libs in a single derivation
+    # nix-tools.exes.cardano-chain for all executables, same for tests and benchmarks.
+    #
+    # The alternative syntax is: nix-tools._raw.cardano-chain.components.$comp
+    # if you want to build only a single component.
```

Comment:
The first use case seems much less common than the second. Usually if I'm building something I want to build a specific thing, not "all libs in this project".

Reply:
The origin came primarily for use cases with `exes`, where you want a single derivation and get all the executables instead of having to write each single executable component to get the relevant component. Behind the scenes it's just a joinSymlink (or alternative logic on windows).

Reply:
Sure, I'm just saying that I do a lot of `nix build -f default.nix myPackage`, so I'd like the syntax for that to be nice.

Reply:
@michaelpj what is `nix build -f default.nix myPackage` supposed to do? You didn't even specify a component? Do you want *all* libs, execs, tests, benchmarks to be built?

Reply:
`nix build -f default.nix myPackage.lib` is also fine. I'd like `nix build -f default.nix myPackage` to be like `cabal build`-ing it, so yeah, build all the components.

Reply:
Is this comment still accurate?

Reply:
Yes. I think this comment is still accurate.